### PR TITLE
Allow to reassign and modify deadline of planned and resolved tasks.

### PR DESCRIPTION
**Pull request by njohner**
Tasks have a separate permission to check whether the edit form is available `opengever.task: Edit task`, this is necessary because certain fields (for example the deadline) could already be modified in review states where the we generally did not want the task to be modifiable, so the edit form was simply protected with `opengever.task: Edit task`. This was not respected in the new UI (see https://4teamwork.atlassian.net/browse/CA-4428) and is fixed here. 
As we now want to allow editing of more fields in various states and for all users with edit permission, we could have simply adapted the `opengever.task: Edit task`, but that allowed to access the edit form, and hence also edit fields such as the task type, which is not desirable. Instead I decided to make two changes to the permissions:
- modify `opengever.task: Edit task` so that editors can modify open tasks (will work in classic and new UI)
- Allow editors to modify the deadline of pending tasks (classic and new UI)
- Allow editors to reassign pending tasks (classic and new UI)
- The rest is done for the new UI only, and is not enforced by the API either:
  - Add an action for editing `relatedItems`
  - Add an action for editing the `text`
 
Editing either the `relatedItems` or `text` will now also generate a response and an activity.

For [CA-4663] and https://4teamwork.atlassian.net/browse/CA-4428

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- Upgrade-Steps:
  - [x] SQL Operations do not use imported model (see [docs](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/994344994/Upgrade-Steps))
  - [ ] Make it deferrable if possible -> not deferrable
  - [x] Execute as much as possible conditionally
- New functionality:
  - [ ] for `task` also works for `forwarding` -> still in discussion, will be handled separately
- New translations
  - [x] All msg-strings are unicode

[CA-4663]: https://4teamwork.atlassian.net/browse/CA-4663?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by phgross — APPROVED** [review]
Nice ✨, LGTM 👍

File: opengever/core/upgrades/20221006085647_allow_editing_pending_tasks/upgrade.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from ftw.upgrade import UpgradeStep
+from opengever.core.upgrade import NightlyWorkflowSecurityUpdater
+
+
+class AllowEditingPendingTasks(UpgradeStep):
+    """Allow editing pending tasks.
+    """
+
```
[the remark this answers is not in the document]
Comment (njohner) [reply opengever/core/upgrades/20221006085647_allow_editing_pending_tasks/upgrade.py:8]: Hum, from what I see we usually do not run those as `deferrable`. As it is important that these upgrade steps are run in the correct order we need to have them either all as `deferrable` or not.

**Comment by njohner** [comment]
@phgross There are 2 new commits. If you have time to have another look...

**Comment by njohner** [comment]
Note that this will also resolve https://4teamwork.atlassian.net/browse/CA-4428

**Review by elioschmutz — APPROVED** [review]
I reviewed the last two commits as @njohner requested.
@phgross already reviewed the previous commits.

**Pull request merged**